src/utils.c

```diff
@@ -0,0 +1,78 @@
+// utils.c
+#include "utils.h"
+#include <limits.h> // For INT16_MIN
+#include <rand.h>
+#include <stdint.h>
+
+inline uint8_t
```

Comment:
I'm not sure inline works from a different compilation unit. Generally inline would be from a header unless you have LTO enabled which may not even apply still.

Reply:
I tested it, and you're right. No difference so have removed the inlines from the file. I've left the abs one in as that is used in a function inside that file.